misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/JDIEventProcessor.java

```diff
@@ -103,11 +104,31 @@ private boolean processEvent(EventSet eventSet, Event event) {
             configureUserBreakPoints(evt.referenceType());
             eventSet.resume();
         } else if (event instanceof BreakpointEvent) {
-            StoppedEventArguments stoppedEventArguments = new StoppedEventArguments();
-            stoppedEventArguments.setReason(StoppedEventArgumentsReason.BREAKPOINT);
-            stoppedEventArguments.setThreadId(((BreakpointEvent) event).thread().uniqueID());
-            stoppedEventArguments.setAllThreadsStopped(true);
-            context.getClient().stopped(stoppedEventArguments);
+            Value value = null;
+            String qualifiedClassName =
+                getQualifiedClassName(context, ((BreakpointEvent) event).location().declaringType());
+            Map<Integer, BalBreakpoint> balBreakpoints = this.balBreakpoints.get(qualifiedClassName);
+            int lineNumber = ((BreakpointEvent) event).location().lineNumber();
+
+            if (balBreakpoints != null && balBreakpoints.containsKey(lineNumber)) {
+                value = context.getAdapter().evaluateExpression(balBreakpoints.get(lineNumber).getCondition(),
+                    ((BreakpointEvent) event).thread());
+            }
+            if (balBreakpoints != null && balBreakpoints.containsKey(lineNumber)
+                && balBreakpoints.get(lineNumber).getCondition() != null
+                && !Boolean.parseBoolean(Objects.requireNonNull(value).toString())) {
```

Comment:
Directly trying to parse into boolean is unsafe, as we cannot guarantee about the type of conditional expression result (can be non-boolean)

Reply:
Fixed